modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/util/RoleNotificationRecipientBuilderUtil.java

```diff
@@ -0,0 +1,274 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.workflow.kaleo.runtime.internal.util;
+
+import com.liferay.depot.constants.DepotRolesConstants;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerList;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerListFactory;
+import com.liferay.osgi.util.service.Snapshot;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Organization;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.model.UserGroupGroupRole;
+import com.liferay.portal.kernel.model.UserGroupRole;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.OrganizationLocalService;
+import com.liferay.portal.kernel.service.UserGroupGroupRoleLocalService;
+import com.liferay.portal.kernel.service.UserGroupRoleLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.workflow.kaleo.definition.NotificationReceptionType;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstanceToken;
+import com.liferay.portal.workflow.kaleo.runtime.ExecutionContext;
+import com.liferay.portal.workflow.kaleo.runtime.notification.NotificationRecipient;
+import com.liferay.portal.workflow.kaleo.runtime.util.validator.GroupAwareRoleValidator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.osgi.framework.BundleContext;
+
+/**
+ * @author Renan Vasconcelos
+ */
+public class RoleNotificationRecipientBuilderUtil {
+
+	public static void addRoleRecipientAddresses(
+			Set<NotificationRecipient> notificationRecipients, Role role,
+			NotificationReceptionType notificationReceptionType,
+			ExecutionContext executionContext)
+		throws Exception {
+
+		List<User> users = _getRoleUsers(role, executionContext);
+
+		for (User user : users) {
+			if (user.isActive()) {
+				NotificationRecipient notificationRecipient =
+					new NotificationRecipient(user, notificationReceptionType);
+
+				notificationRecipients.add(notificationRecipient);
+			}
+		}
+	}
+
+	public static void deactivate() {
+		_serviceTrackerList.close();
+	}
+
+	public static ServiceTrackerList<GroupAwareRoleValidator>
+		setServiceTrackerList(BundleContext bundleContext) {
```

Comment:
@RenanUmV It's a bad pattern to start/stop the service tracker list by another class, because as a util your other methods are always available to all who can see the util class.